apps/shared/models/server-models.ts

```diff
@@ -0,0 +1,68 @@
+import {firestore} from 'firebase-admin';
+import {
+  QueryDocumentSnapshot as AdminQueryDocumentSnapshot,
+  FirestoreDataConverter as AdminFirestoreDataConverter,
+  DocumentReference as AdminDocumentReference,
+  DocumentData as AdminDocumentData,
+} from '@google-cloud/firestore';
+
+import {GithubUser} from './user';
+import {GithubLabel} from './label';
+import {GithubMilestone} from './milestone';
+import {GithubPullRequest} from './pull-request';
+import {GithubStatus} from './status';
+import {GithubTeam} from './team';
+
+import {Constructor, BaseModel, GithubHelperFunctions, GithubBaseModel} from './base';
+import {GithubCheck} from './check';
+export const User = forServer(GithubUser, GithubUser.githubHelpers);
+export const Label = forServer(GithubLabel, GithubLabel.githubHelpers);
+export const Milestone = forServer(GithubMilestone, GithubMilestone.githubHelpers);
+export const PullRequest = forServer(GithubPullRequest, GithubPullRequest.githubHelpers);
+export const Status = forServer(GithubStatus, GithubStatus.githubHelpers);
+export const Team = forServer(GithubTeam, GithubTeam.githubHelpers);
+export const Check = forServer(GithubCheck, GithubCheck.githubHelpers);
+
+type WithAdminFirestoreDataConverter<Model> = Model & {
+  converter: AdminFirestoreDataConverter<Model>;
+};
+type WithAdminFirestoreDataConverterAndGithub<Model, GithubModel> = Model & {
+  converter: AdminFirestoreDataConverter<Model> & {
+    fromGithub: (model: GithubModel) => Model;
+    getFirestoreRefForGithubModel: (model: GithubModel) => AdminDocumentReference;
+  };
+};
+
+export function forServer<
```

Comment:
I like the architecture you planned for the models. It looks great, and actually is very easy to follow. I think we could make this perfect by having small little comments w/ TSDoc for `forApp`, `forServer` etc. wdyt?